news-bundle/src/EventListener/DataContainer/PageListener.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\NewsBundle\EventListener\DataContainer;
+
+use Contao\CoreBundle\Framework\ContaoFramework;
+use Contao\CoreBundle\ServiceAnnotation\Callback;
+use Contao\CoreBundle\ServiceAnnotation\Hook;
+use Contao\NewsArchiveModel;
+use Contao\NewsBundle\Controller\NewsFeedController;
+use Symfony\Component\Security\Core\Security;
+
+class PageListener
+{
+    public function __construct(private ContaoFramework $framework, private Security $security)
+    {
+    }
+
+    /**
+     * @Callback(table="tl_page", target="fields.archives.options")
+     */
+    public function getAllowedArchives(): array
+    {
+        $model = $this->framework->getAdapter(NewsArchiveModel::class);
+        $user = $this->security->getUser();
+
+        $archive = $user->isAdmin ? $model->findAll() : $model->findMultipleByIds($user->news);
```

Comment:
```suggestion
        if (!$user instanceof BackendUser || $this->security->isGranted('ROLE_ADMIN')) {
            $archive = $model->findAll();
        } else {
            $archive = $model->findMultipleByIds($user->news);
        }
```
(might need `use Contao\BackendUser`)